pandas/tests/arrays/test_datetimes.py

```diff
@@ -16,6 +16,25 @@
 
 
 class TestDatetimeArrayConstructor(object):
+    def test_mixing_naive_tzaware_raises(self):
+        # GH#24569
+        arr = np.array([pd.Timestamp('2000'), pd.Timestamp('2000', tz='CET')])
```

Comment:
Is there a test that checks that verifies mixed tz example from the issue raises?  I didn't see one quickly glancing over this file, but could be missing it or it could be elsewhere.  If not, you could maybe parametrize over a `data` variable that does both orders for aware/naive and the aware/aware case and get ride of the `for obj` loop, i.e.

```python
@pytest.mark.parametrize('data', [
    [pd.Timestamp('2000', tz='CET'), pd.Timestamp('2000')],
    [pd.Timestamp('2000'), pd.Timestamp('2000', tz='CET')],
    [pd.Timestamp('2000', tz="US/Central"), pd.Timestamp('2000', tz='CET')]])
def test_mixing_naive_tzaware_raises(self):
    arr = np.array(data)
```

Reply:
I think this does does hit the case from this original issue, in the `DatetimeArray._from_sequence` case.

Reply:
Yes, agreed that the `DatetimeArray._from_sequence` case hits the original issue.

I think my comment was a bit unclear; what I was asking is if we have coverage for the working mixed tz example you also provided in the issue: `mix = np.array([pd.Timestamp('2000', tz="US/Central"), pd.Timestamp('2000', tz='CET')])`.  I didn't see a test for that anywhere, and figured we could shoehorn it into this test if it's indeed not covered.

Reply:
Not a pressing issue or anything though, so if it's not covered it we could just make it a follow-up item; don't want this to block other work.